_assets/javascripts/lib/distance-sorter.js

```diff
@@ -0,0 +1,167 @@
+/* eslist no-plusplus: 0 */
+/* global CRDS */
+/* jshint esversion: 6 */
+
+window.CRDS = window.CRDS || {};
+
+CRDS.DistanceSorter = class DistanceSorter {
+  constructor() {
+    this.locationDistances = [];
+    this.searchForm = undefined;
+    this.searchInput = undefined;
+    this.formSubmitButton = undefined;
+    this.cards = undefined;
+    this.locationFinder = undefined;
+    this.init();
+  }
+
+  init() {
+    this.searchForm = document.getElementById('locations-address-input');
+    this.searchForm.addEventListener('submit', this.handleFormSubmit.bind(this));
+    this.searchInput = this.searchForm.getElementsByTagName('input')[0];
+    this.formSubmit = this.searchForm.getElementsByTagName('button')[0];
+    this.formSubmit.disabled = false;
+    this.locationsCarousel = DistanceSorter.getLocationsCarousel();
+    this.cards = this.locationsCarousel.cards;
+    this.locationFinder = new CRDS.LocationFinder();
+  }
+
+  static getLocationsCarousel() {
+    return Object.values(CRDS._instances).find(
+      (instance) => DistanceSorter._isSectionLocationsCarousel(instance)
+    );
+  }
+
+  static _isSectionLocationsCarousel(instance) {
+    return instance.carousel !== undefined && instance.carousel.id === 'section-locations';
```

Comment:
One small improvement here would be to switch from an `id` to something like a `data-attribute`. I think it could make this a little less brittle...